docs: mention netlify permalinks in release publishing guide (#664)
cal-itp/calitp.org: docs/guides/dsdl/release.md

The steps to create a release tag are identical to [Benefits](https://docs.calitp.org/benefits/guides/release/). Release candidate tags are skipped.

> After a release has been published it is necessary to identify the associated netlify deployment and use the permalink to update the list below.

Reviewer: @jgravois  - re: https://github.com/cal-itp/calitp.org/issues/662#issuecomment-4712842834,

> when i sat down and looked at this it felt like a more helpful v1 would be to just let folks copy/paste the entire link tag out of the documentation instead of displaying one generic snippet and then appending a lookup table and asking folks to inject the url itself from there.

Yeah, I can see how it kinda feels premature to write this page like we're gonna have a bunch of versions.

Maybe the snippet could always have the latest URL? And the table or list is where the older versions would go.

I'm ok with not having the table or list right now, and if that's the case, could we re-word the last part of this sentence? Maybe something like:

```suggestion
After a release has been published, it is necessary to identify the associated Netlify deployment and document the release's permalink on this page.
```

Author: these are all great suggestions 🤙
